posthog/hogql/metadata.py

```diff
@@ -138,3 +142,19 @@ def is_valid_view(select_query: ast.SelectQuery | ast.SelectSetQuery) -> bool:
                 if field.chain and field.chain[-1] == "*":
                     return False
     return True
+
+
+def get_table_names(select_query: ast.SelectQuery | ast.SelectSetQuery) -> list[str]:
+    # Don't need types, we're only interested in the table names as passed in
+    collector = TableCollector()
+    collector.visit(select_query)
+    return list(collector.table_names)
+
+
+class TableCollector(TraversingVisitor):
+    def __init__(self):
+        self.table_names = set()
+
+    def visit_join_expr(self, node: ast.JoinExpr):
+        if isinstance(node.table, ast.Field):
+            self.table_names.add(node.table.chain[0])
```

Comment:
Can we get some tests on this please - I don't think this is gonna be perfect, e.g. use this with CTEs

Reply:
yep forgot about that, good call. just added a bunch and updated the implementation